frame/contracts/src/wasm/runtime.rs

```diff
@@ -2395,11 +2395,17 @@ pub mod env {
 		str_len: u32,
 	) -> Result<ReturnCode, TrapReason> {
 		ctx.charge_gas(RuntimeCosts::DebugMessage)?;
-		if ctx.ext.append_debug_buffer("") {
+		if ctx
+			.ext
+			.append_debug_buffer("")
+			.map_err(|err| TrapReason::SupervisorError(err))?
```

Comment:
`map_err` is not needed. There is a `impl From<DispatchError> for TrapReason` that takes care of this. A simple `?` is enough.